packages/text-field/index.tsx

```diff
@@ -284,7 +284,15 @@ class TextField<
     const {props} = child;
     return Object.assign({}, props, {
       foundation: this.state.foundation,
-      handleFocusChange: (isFocused: boolean) => this.setState({isFocused}),
+      handleFocusChange: (isFocused: boolean) => {
+        this.setState({isFocused});
+        if (!this.state.foundation) return;
+        if (isFocused) {
+          this.state.foundation.activateFocus();
```

Comment:
Why do we have to have foundation as state? We can make text field not accept foundation prop to keep it simple.

Reply:
we actually depend on the foundation in the `<Input />` component, and sometimes require a rerender when the foundation updates or is created -- hence the state. Are you saying to not have `<Input />` accept the `foundation` prop?